test/jdk/java/security/KeyPairGenerator/SolarisShortDSA.java

```diff
@@ -67,4 +71,13 @@ static boolean use(KeyPair kp) throws Exception {
         sig2.update(data);
         return sig2.verify(signed);
    }
+
+   private static int xBitLength(PrivateKey priv, Provider provider) throws Exception {
+        if (priv instanceof DSAPrivateKey) {
+            return ((DSAPrivateKey) priv).getX().bitLength();
+        }
+        KeyFactory kf = KeyFactory.getInstance("DSA", provider);
+        DSAPrivateKeySpec spec = kf.getKeySpec(priv, DSAPrivateKeySpec.class);
+        return spec.getX().bitLength();
+    }
```

Comment:
Two questions:
* Why are we concerned about this test which seems specific to Solaris, which is a platform not supported by Semeru?
* How does it make sense to ignore the specific key to answer a question about that key?

Reply:
My second question is the more important one: How does it make sense to ignore the specific key to answer a question about that key?